lib/db/task.ts

```diff
@@ -0,0 +1,114 @@
+import { start } from "repl";
+import prisma from "./prisma";
+import { Task } from "@prisma/client";
+
+/**
+ * Retrieves a single task based on provided parameters.
+ *
+ * @export
+ * @async
+ * @param {Partial<Task>} params Information about the task.
+ * @returns {Promise<Task | null>}
+ *} The retrieved task object.
+ */
+export async function getTask(params: Partial<Task>): Promise<Task | null> {
+  const task = await prisma.task.findFirst({
+    where: {
+      ...params,
+    },
+    include: {
+      user: true,
+      activities: true,
+      comments: true,
+      status: true,
+      sprint: true,
+    },
+  });
+  return task;
+}
+
+/**
+ * Retrieves multiple tasks based on provided parameters.
+ *
+ * @export
+ * @async
+ * @param {Partial<Task>} params Information about the tasks to retrieve.
+ * @returns {Promise<Task[]>} An array of retrieved task objects.
+ */
+export async function getTasks(params: Partial<Task>): Promise<Task[]> {
+  const task = await prisma.task.findMany({
+    where: {
+      ...params,
+    },
+    include: {
+      user: true,
+      activities: true,
+      comments: true,
+      status: true,
+      sprint: true,
+    },
+  });
+  return task;
+}
+
+/**
+ * Creates a new task.
+ *
+ * @export
+ * @async
+ * @param {Partial<Task>} params Information about the task to create.
+ * @returns {Promise<Task>} The created task object.
+ */
+export async function createTask(params: any): Promise<Task> {
+  const task = await prisma.task.create({
+    data: {
```

Comment:
Same note as in `sprint.ts`: can spread syntax be used here?

Reply:
@winstuhn

Reply:
I'll keep looking for a work-around, for now, I am unable to use spread syntax without it causing errors.